internal/metrics/metrics.go

```diff
@@ -7,34 +7,33 @@ import (
 	"github.com/pkg/errors"
 )
 
-// Query is the filter used to retrieve metrics data.
-type Query interface {
-	Query() string
-}
-
 // AlignReduce is the type to enumerate allowed combinations of per series
 // aligner and cross series reducer.
 type AlignReduce int32
 
 // Series aligner and cross series reducer types (for latency).
 const (
-	Align99Reduce99 AlignReduce = 1
-	Align95Reduce95             = 2
-	Align50Reduce50             = 3
+	Align99Reduce99 AlignReduce = iota + 1
+	Align95Reduce95
+	Align50Reduce50
 )
 
 // Provider represents a metrics Provider such as Stackdriver.
 type Provider interface {
+	// Sets the candidate revision name for which the provider should get
+	// metrics.
+	SetCandidateRevision(revisionName string)
```

Comment:
Why is this not part of constructor of each provider? I don't see a need to change interface. The same can apply to other filters, but you have revisionName singled out in a new method?

Reply:
We are injecting the metrics provider into the `rollout.Rollout` struct when we initialize it with `rollout.New`. At this point, we still don't know the candidate revision since it is determine during the [rollout process](https://github.com/GoogleCloudPlatform/cloud-run-release-operator/blob/main/pkg/rollout/rollout.go#L93). 

If we were to remove this function from the interface, we'd have to do all that logic about getting the stable and candidate revisions before initializing `rollout.Rollout`.

Reply:
Let's add a todo like:

> TODO(gvso) Consider removing this method and making revisionName part of other method signatures.

Reply:
Done!